patches/minecraft/net/minecraft/block/Block.java.patch

```diff
@@ -1354,12 +1354,25 @@
 +        return func_185467_w();
 +    }
 +
++    /**
++     * Gets the color that the beacon beam should be blended with.
++     *
++     * This should return null for no change, and a three entry float array with RGB values from 0.0 to 1.0.
++     *
++     * @param world The world the block is in
++     * @param blockState The state of the block
++     * @return The float array of colors to blend
++     * */
++    public float[] getBeaconColor(World world, IBlockState blockState) {
++        return null;
```

Comment:
I did have a length and null check, should I do bounds checking?

On Aug 4, 2016 1:42 PM, "Henry Loenwind" <notifications@github.com> wrote:

> In patches/minecraft/net/minecraft/block/Block.java.patch
> <https://github.com/MinecraftForge/MinecraftForge/pull/3159#discussion_r73579651>
> :
>
> > @@ -1354,12 +1354,25 @@
> >  +        return func_185467_w();
> >  +    }
> >  +
> > ++    /**
> > ++     * Gets the color that the beacon beam should be blended with.
> > ++     *
> > ++     * This should return null for no change, and a three entry float array with RGB values from 0.0 to 1.0.
> > ++     *
> > ++     * @param world The world the block is in
> > ++     * @param blockState The state of the block
> > ++     * @return The float array of colors to blend
> > ++     * */
> > ++    public float[] getBeaconColor(World world, IBlockState blockState) {
> > ++        return null;
>
> On the one hand that float[] is exactly what mc then needs to process the
> color, on the other hand it provides very little type safety. I'm a bit
> torn here, should I recommend you use a better return value or not?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/MinecraftForge/MinecraftForge/pull/3159/files/c4cdbb020c7a9be3055a2c59fe873c7bcd83b43c#r73579651>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AFYYWeu74ugM9-dmeXEgzSa55YVsxr4eks5qcjKQgaJpZM4Jb-cJ>
> .
>